### PR TITLE
Adding PR submission section to contributors guide

### PR DESCRIPTION
**Pull request by pmccarthy**
**What**
Adding new PR submission section to the contributors guide. This section is purposely high level as we'll most likely want to add some component specific guidelines in each repository

### CODE REVIEW

**Review by philbrookes — APPROVED** [review]
This looks good to me, although "no issue required" does appear to conflict with the PR workflow that's trying to be added to ensure every PR has a linked issue?

e.g. the one that's failing on this PR 😂

File: src/contributing.md [hunk 1]
```diff
@@ -52,6 +52,14 @@ A list of good first issues can be found from the Kuadrant Github [projects boar
...
 
+## Submitting Pull Requests
+
+When submitting a pull request against one of the Kuadrant component repositories, should the PR address a particular Github issue please make sure to reference it in the PR description. That said, it is not mandatory for a PR to have an associated issue referenced. In the event of a standalone PR that doesn't have an associated issue, please add a detailed description of the changes included. Adding *What* and *Why* sections is a good start. For example, *What* is the purpose of this change and *Why* is it required and/or being implemented in this way?
+
+The Kuadrant project owners and maintainers strive to review and/or respond to all newly submitted PRs in a timely manner, however, if you're finding it difficult to get someone to review your PR, please reach out directly on [Slack][SlackChannelURL] or [mail][MailingList]
+
+Finally, it is recommended that you squash your changes into a single commit where possible. If this is not feasible please ensure that your commits are representing a logical piece of work that can be reviewed independently within the PR.
```
Comment (guicassolato) [inline src/contributing.md:61]: In some repos, we also require [signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits). E.g. https://docs.kuadrant.io/authorino/docs/contributing/#sign-your-commits

**Review by guicassolato — APPROVED** [review]
/lgtm

**Comment by guicassolato** [comment]
> Hmmm..I thought the purpose of that job was to ensure the PR and/or issue is showing on our project board? Rather than ensuring that there's an associated issue? Perhaps I'm wrong on that?

You are correct, @pmccarthy .

**Comment by guicassolato** [comment]
> > > Hmmm..I thought the purpose of that job was to ensure the PR and/or issue is showing on our project board? Rather than ensuring that there's an associated issue? Perhaps I'm wrong on that?
> > 
> > 
> > You are correct, @pmccarthy .
> 
> oh, interesting, so if there is no linked issue, this job should not fail? I feel like it does currently.

We're still investigating why the job is failing, but I guess it has nothing to do with having or not having an issue linked to the PR. The PR itself is a kind of GitHub issue. AFAIK the job is only to set the **Kuadrant** project on all kinds of issues (actual issues or PRs.)

**Pull request merged**